catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/event/SubscriptionManager.java

```diff
@@ -0,0 +1,345 @@
+
+import org.apache.commons.lang.StringUtils;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.cm.Configuration;
+import org.osgi.service.cm.ConfigurationAdmin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.event.InvalidSubscriptionException;
+import ddf.catalog.event.Subscription;
+
+/**
+ * SubscriptionManager manages registeredSubscriptions for add, delete, and persistence.
+ */
+public class SubscriptionManager {
+    /**
+     * The log4j BASE_LOGGER for this class
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubscriptionManager.class);
+
+    /**
+     * The ManagedServiceFactory PID for the subscription's callback web service adapter,
+     * used to dynamically instantiate the web service's adapter
+     */
+    private static final String FACTORY_PID = "CSW_Subscription";
+
+    private static Map<String, ServiceRegistration<Subscription>> registeredSubscriptions =
+            new HashMap<>();
+
+    private static Map<String, Subscription> subscriptions = new HashMap<>();
+
+    protected SubscriptionManager() {
+        // SubscriptionManager should be used statically
+        throw new UnsupportedOperationException();
+    }
+
+    public static boolean hasSubscription(String subscriptionId) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("subscriptionUuid = " + subscriptionId);
+        }
+
+        return registeredSubscriptions.containsKey(subscriptionId);
+    }
+
+    /**
+     * Concatenate the subscription Id and deliveryMethodUrl to create subscriptionUuid.
+     *
+     * @param subscriptionId - in String format
+     * @return a subscriptionUuid filter in String format
+     * @parm deliveryMethodUrl
+     * - in String format
+     */
+    public static String getSubscriptionUuid(String subscriptionId) {
+        if (subscriptionId == null || subscriptionId.isEmpty()) {
+            return UUID.randomUUID()
+                    .toString();
+        }
+        return subscriptionId;
+    }
+
+    public static Subscription getSubscription(String subscriptionId) {
+        return subscriptions.get(subscriptionId);
+        // TODO (RWY) - rather than have a seperate hashmap can we do something similar to getSubscriptionConfiguration
+    }
+
+    public static String updateSubscription(BundleContext context,  Subscription sub,
+            String subscriptionId, String deliveryMethodUrl, boolean persistSubscription)
+            throws InvalidSubscriptionException {
+        if (!hasSubscription(subscriptionId)) {
+            throw new InvalidSubscriptionException(
+                    "Unable to create subscription because subscription ID was not found");
+        }
+        return addOrUpdateSubscription(context, sub, subscriptionId, deliveryMethodUrl, false);
```

Comment:
Should you pass in `persistSubscription` instead? If not, then maybe it can be removed from the method's signature.